### ISSUE TITLE
The model will diverge at different epochs depending on the settings n_epochs

### ISSUE THREAD

**Issue by Shaunixiayu** [junyanz/pytorch-CycleGAN-and-pix2pix#1699]
Why I set the number of epochs to 100, when the model runs to the 100th epoch will diverge, its cycle loss all become very large, when the number of epochs is set to 50, it will diverge after 50 epochs, this is the reason for the learning rate decay, I have modified many times still can not solve this problem, I hope you can help me explain, thank you very much, the following is my training command: =========================
python train.py --dataroot./datasets/my_dataset --name test_digital_core --model cycle_gan -- batch_size 2 -- load_size 550 -- crop_size 500
 --NetG resnet_9blocks -- num_threads 8 -- n_epochs 100 -- n_epochs_decay 100 -- display_freq 1000 -- save_latest_freq 10000 -- gpu_ids 0 -- input_nc 1 -- output_nc 1 -- lr 0.00005 -- lambda_identity 0.5 -- display_id 0 -- lr_policy linear -- lr_decay_iters 50
=======================
Here is a snippet of loss_log:
(Epoch: 100, iters: 450, time: 0.253, data: 0.003) D_A: 0.186 G_A: 0.381 cycle_A: 0.232 idt_A: 0.083 D_B: 0.156 G_B: 0.327 cycle_B: 0.279 idt_B: 0.052
(Epoch: 100, iters: 550, time: 1.679, data: 0.003) D_A: 0.208 G_A: 0.352 cycle_A: 0.216 idt_A: 0.080 D_B: 0.168 G_B: 0.453 cycle_B: 0.270 idt_B: 0.049
(Epoch: 101, iters: 100, time: 0.253, data: 0.309) D_A: 61.812 G_A: 108.169 cycle_A: 0.783 idt_A: 0.103 D_B: 86.232 G_B: 150.874 cycle_B: 1.823 idt_B: 0.156
(Epoch: 101, iters: 200, time: 0.248, data: 0.003) D_A: 1148.972 G_A: 1809.694 cycle_A: 2.853 idt_A: 0.198 D_B: 1108.889 G_B: 1652.918 cycle_B: 6.601 idt_B: 0.166
(Epoch: 101, iters: 300, time: 0.252, data: 0.007) D_A: 1772.589 G_A: 2515.135 cycle_A: 3.453 idt_A: 0.199 D_B: 1788.356 G_B: 2537.567 cycle_B: 6.494 idt_B: 0.154
(Epoch: 101, iters: 400, time: 0.253, data: 0.002) D_A: 2305.573 G_A: 3069.433 cycle_A: 3.639 idt_A: 0.174 D_B: 2186.480 G_B: 2944.759 cycle_B: 7.144 idt_B: 0.139
(Epoch: 101, iters: 500, time: 0.253, data: 0.002) D_A: 2599.390 G_A: 3456.715 cycle_A: 5.050 idt_A: 0.191 D_B: 2767.597 G_B: 3279.008 cycle_B: 6.658 idt_B: 0.180
(Epoch: 102, iters: 50, time: 0.253, data: 0.018) D_A: 3118.727 G_A: 4030.425 cycle_A: 4.975 idt_A: 0.196 D_B: 3209.157 G_B: 3905.536 cycle_B: 5.776 idt_B: 0.164
==========================

**Comment by junyanz** [comment]
Which version/code are you using? Have you tried training the model with our default dataset and see if it works?

**Comment by Shaunixiayu** [comment]
> Which version/code are you using? Have you tried training the model with our default dataset and see if it works?您使用的是哪个版本/代码？您是否尝试过使用我们的默认数据集训练模型，看看它是否有效？

Thank you very much for your reply. I solved the problem by running the default dataset due to an error caused by accidentally copying the code that updated the learning rate.
I can now run my dataset normally. The problem now is that the performance of the model is not good enough. My trainA has more than 400 graphs, and so does trainB. The epoch runs about 150, but the performance of the discriminator and generator does not seem to be very good, as shown below:
(Epoch: 180, iters: 50, time: 0.241, data: 0.004) D_A: 0.294 G_A: 0.277 cycle_A: 0.159 idt_A: 0.116 D_B: 0.305 G_B: 0.223 cycle_B: 0.259 idt_B: 0.055
(Epoch: 180, iters: 150, time: 0.242, data: 0.002) D_A: 0.257 G_A: 0.231 cycle_A: 0.166 idt_A: 0.098 D_B: 0.237 G_B: 0.284 cycle_B: 0.246 idt_B: 0.069
(Epoch: 180, iters: 250, time: 0.242, data: 0.003) D_A: 0.223 G_A: 0.317 cycle_A: 0.207 idt_A: 0.091 D_B: 0.266 G_B: 0.314 cycle_B: 0.243 idt_B: 0.062
(Epoch: 180, iters: 350, time: 0.241, data: 0.002) D_A: 0.206 G_A: 0.293 cycle_A: 0.224 idt_A: 0.088 D_B: 0.259 G_B: 0.393 cycle_B: 0.254 idt_B: 0.064
Do you have any good suggestions? What should I do to make the model perform better?
